storage/remote/queue_manager.go

```diff
@@ -449,11 +457,16 @@ func (s *shards) runShard(i int) {
 			for len(pendingSamples) >= s.qm.cfg.MaxSamplesPerSend {
 				s.sendSamples(pendingSamples[:s.qm.cfg.MaxSamplesPerSend])
 				pendingSamples = pendingSamples[s.qm.cfg.MaxSamplesPerSend:]
+
+				if !timer.Stop() {
```

Comment:
DRY - these six lines are the same as the `func()` you have defined at the top - just name it and call it.

Reply:
It occurred to me we don't really want to do this in a loop, but then I realised the loop can never execute more than once since we add samples one at a time.  Maybe it shouldn't be a loop, or should be commented as a special kind of loop.

Reply:
I made it a special kind of loop - an `if` loop.

Reply:
I was talking about the `for len(pendingSamples) >= s.qm.cfg.MaxSamplesPerSend` loop

Reply:
oops, see what you mean